src/server/firefox.ts

```diff
@@ -151,88 +147,59 @@ function wrapTransportWithWebSocket(transport: ConnectionTransport, logger: Inne
         pendingBrowserContextDeletions.delete(seqNum);
       }
 
-      message.id = id;
-      socket.send(JSON.stringify(message));
+      source.send(JSON.stringify(message));
       return;
-    }
-
-    // Process notification response.
-    const { method, params, sessionId } = message;
-    if (sessionId) {
-      const socket = sessionToSocket.get(sessionId);
-      if (!socket || socket.readyState === ws.CLOSING) {
-        // Drop unattributed messages on the floor.
+    },
+
+    onBrowserNotification(message: ProtocolResponse) {
+      // Process notification response.
+      const { method, params, sessionId } = message;
+      if (sessionId) {
+        const data = sessionToData.get(sessionId);
+        if (!data || data.socket.readyState === ws.CLOSING) {
+          // Drop unattributed messages on the floor.
+          return;
+        }
+        data.socket.send(JSON.stringify(message));
         return;
       }
-      socket.send(JSON.stringify(message));
-      return;
-    }
-    if (method === 'Browser.attachedToTarget') {
-      const socket = browserContextIds.get(params.targetInfo.browserContextId);
-      if (!socket || socket.readyState === ws.CLOSING) {
-        // Drop unattributed messages on the floor.
+      if (method === 'Browser.attachedToTarget') {
+        const socket = browserContextIds.get(params.targetInfo.browserContextId);
+        if (!socket || socket.readyState === ws.CLOSING) {
+          // Drop unattributed messages on the floor.
```

Comment:
Should we detach in this case?

Reply:
Socket is closing, which means we will run through the close sequence, or what do you mean?